rust/src/smtp/mod.rs

```diff
@@ -0,0 +1,18 @@
+/* Copyright (C) 2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+pub mod smtp;
```

Comment:
Nothing to do with the code, but Rust style rules don't like creating the scenario of smtp/mod.rs with a contained module of the same name (smtp/smtp.rs).

The specific lint is: https://rust-lang.github.io/rust-clippy/master/

We're pretty bad for this in our code (or maybe we shouldn't consider it bad?).  But just a thought, that if this group of functions can be described better then just `smtp`, maybe they should get a different module name?

Reply:
Sure. I can certainly rename this one.. this is kind of not even in accordance with our existing scheme either. It's not a full parser.. and it's gonna be just some helper fns purely written in Rust.